File: lab9KononKI305/phone_menu.py
# phone_menu.py
class PhoneMenu:
    def __init__(self, phone):
        self.phone = phone

    def display_menu(self):
        choice = 0

        while choice != 6:
            print("\nMenu:")
            print("1. Make a call")
            print("2. Send a message")
            print("3. Display call history")
            print("4. Display messages")
            print("5. Display full phone information")
            print("6. Power off")

            try:
                choice = int(input("Enter your choice (1-6): "))
                if choice == 1:
                    self.phone.make_call_user_input()
                elif choice == 2:
                    self.phone.send_message_user_input()
                elif choice == 3:
                    self.phone.display_call_history()
                elif choice == 4:
                    self.phone.display_messages()
                elif choice == 5:
                    self.phone.display_full_info()
                elif choice == 6:
                    self.phone.power_off()
                else:
                    print("Invalid choice. Please enter a number between 1 and 6.")
            except ValueError:
                print("Invalid input. Please enter a valid number.")

File: lab3KononKI305/src/PhoneMenu.java
// PhoneMenu.java
import java.util.Scanner;

public class PhoneMenu {
    private Phone phone;

    public PhoneMenu(Phone phone) {
        this.phone = phone;
    }

    public void displayMenu() {
        Scanner scanner = new Scanner(System.in);
        int choice;

        do {
            System.out.println("\nMenu:");
            System.out.println("1. Make a call");
            System.out.println("2. Send a message");
            System.out.println("3. Display call history");
            System.out.println("4. Display messages");
            System.out.println("5. Display full phone information");
            System.out.println("6. Power off");
            System.out.print("Enter your choice (1-6): ");
            choice = scanner.nextInt();

            switch (choice) {
                case 1:
                    phone.makeCall();
                    break;
                case 2:
                    phone.sendMessage();
                    break;
                case 3:
                    phone.displayCallHistory();
                    break;
                case 4:
                    phone.displayMessages();
                    break;
                case 5:
                    phone.displayFullInfo();
                    break;
                case 6:
                    phone.powerOff();
                    break;
                default:
                    System.out.println("Invalid choice. Please enter a number between 1 and 6.");
            }
        } while (choice != 6);
    }
}
